autonomi/src/network/local.rs

```diff
@@ -0,0 +1,173 @@
+use ant_networking::find_local_ip;
+use anyhow::{Context, Result};
+use libp2p::Multiaddr;
+use std::net::UdpSocket;
+use std::process::Stdio;
+use std::time::Duration;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::process::{Child, Command};
+use tokio::time::sleep;
+
+pub struct LocalNode {
+    child: Child,
+    multiaddr: Multiaddr,
+}
+
+impl LocalNode {
+    pub async fn start() -> Result<Self> {
+        Self::start_internal(true).await
+    }
+
+    pub async fn start_secondary() -> Result<Self> {
+        Self::start_internal(false).await
+    }
```

Comment:
I'm not sure what `start` and `start_secondary` involves here, could this be precised?